### PR TITLE
Prefer destination info over everything else

### PR DESCRIPTION
**Pull request by bsudekum**
We we're preferring name/ref ceremonial hack over destination information. We should show destination information since we'll already get the shield representing the highway number on the right.

Also, I removed the `numberOfDestinationLines` of math for destinations. This is because we want the destination info in the UI to match real world 1:1. Perhaps there should be some guard in place so this does not get out of hand. However, comparing this to waze, they put even more than just destination info. They also stuff in exit information when available.

/cc @frederoni @1ec5 @ericrwolfe

### CODE REVIEW

**Comment by bsudekum** [comment]
@1ec5 this is ready for review

File: MapboxNavigation/Resources/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -19,6 +19,9 @@
 /* Delimiter between route numbers in a road concurrency */
 "REF_DELIMITER" = " / ";
 
+/* Delimiter between destinations in a road concurrency */
```
Comment (1ec5) [inline MapboxNavigation/Resources/Base.lproj/Localizable.strings]: A road concurrency is when a stretch of road is part of multiple routes. This delimiter is merely used when there are multiple destinations, so not necessarily a road concurrency.

File: MapboxNavigation/RouteManeuverViewController.swift [hunk 2]
```diff
@@ -170,12 +170,12 @@ class RouteManeuverViewController: UIViewController {
     }
     
     func updateStreetNameForStep() {
-        if let step = step, step.isNumberedMotorway, let codes = step.codes {
+        if let destinations = step?.destinations {
+            destinationLabel.unabridgedText = destinations.joined(separator: NSLocalizedString("DESTINATION_DELIMITER", bundle: .mapboxNavigation, value: " / ", comment: "Delimiter between destinations in a road concurrency"))
```
Comment (1ec5) [inline MapboxNavigation/RouteManeuverViewController.swift]: We should still find a way to ensure that a destination doesn’t get cut off in the middle, like “Stevens Creek Blvd / West San ~~Carlos Street~~”.

Comment (1ec5) [inline MapboxNavigation/RouteManeuverViewController.swift]: This change could cause the label’s font size to shrink beyond what it normally used to. At some point, would it be more readable to allow a second line of destination text?

Comment (bsudekum) [inline MapboxNavigation/RouteManeuverViewController.swift]: Here is how small it gets and how much we can stuff in there:

simulator screen shot jul 25 2017 11 18 19 am

I think this is okay.

**Comment by bsudekum** [comment]
@1ec5 I think the truncation will be okay. We can adjust the look of many destinations when we refactor this view.

**Comment by 1ec5** [comment]
Mind adding a snapshot test or two so we get a better sense of how this looks in practice?

**Comment by bsudekum** [comment]
@1ec5 updated and also enabled tests and updated old fixtures.

**Comment by bsudekum** [comment]
@1ec5 alright, this is ready I believe

**Comment by frederoni** [comment]
Nit: The snapshots are only tested on iPhone 6S Plus running iOS 10.3.1 so there's no need to add 2x reference images or for other versions of iOS.

**Pull request merged**